addon/session-stores/cookie.js

```diff
@@ -107,6 +143,21 @@ export default BaseStore.extend({
   */
   persist(data) {
     this._lastData = data;
+    if (data.hasOwnProperty('cookieDomain')) {
+      this.set('cookieDomain', data.cookieDomain);
+      delete data.cookieDomain;
+    }
+
+    if (data.hasOwnProperty('cookieName')) {
+      this.set('cookieName', data.cookieName);
+      delete data.cookieName;
+    }
+
+    if (data.hasOwnProperty('cookieExpirationTime')) {
+      this.set('cookieExpirationTime', data.cookieExpirationTime);
+      delete data.cookieExpirationTime;
+    }
+
```

Comment:
When would `data` contain these options (`cookieDomain` etc.)? The idea would be that `data` only contains the actual payload to write to the cookie and the options for `cookieDomain` etc. are just read from the instance itself (`this.get('cookieDomain')` etc.).